client/dashboard/app/queries/domain-whois.ts

```diff
@@ -0,0 +1,35 @@
+import { mutationOptions, queryOptions } from '@tanstack/react-query';
+import {
+	fetchDomainWhois,
+	validateDomainWhois,
+	updateDomainWhois,
+	type DomainContactDetails,
+} from '../../data/domain-whois';
+import { queryClient } from '../query-client';
+
+export const domainWhoisQuery = ( domainName: string ) =>
+	queryOptions( {
+		queryKey: [ 'domains', domainName, 'whois' ],
+		queryFn: () => fetchDomainWhois( domainName ),
+	} );
+
+export const domainWhoisValidateQuery = (
+	domainName: string,
+	domainContactDetails: DomainContactDetails
+) =>
+	queryOptions( {
+		queryKey: [ 'domains', domainName, 'whois', 'validate' ],
+		queryFn: () => validateDomainWhois( domainName, domainContactDetails ),
+	} );
+
+export const domainWhoisMutation = ( domainName: string ) =>
+	mutationOptions( {
+		mutationFn: ( {
+			formData,
+			transferLock,
+		}: {
+			formData: DomainContactDetails;
```

Comment:
[SUPER nit]: I think it's better to call this `domainContactDetails` to match with the parameter in the function declaration. Besides "form data" is a concept in the component only.